File: src/main/java/com/tom/Hello.java
package com.tom;

public class Hello {
    public static void main(String[] args) {
//        System.out.println("Hello World!");

        String s = "abcde";
        System.out.println(s.charAt(3)); //index=3的字
        System.out.println(s.length());  //總長度
        System.out.println(s.substring(3));  //取子字串從index=3開始取到最後
        System.out.println(s.substring(1,4));  //取子字串從index=1開始取到index=4

        Person p = new Person("A",45f,1.6f);
        p.hello();
       /* p.weight = 45f;
        p.height = 1.6f;*/
        System.out.println(p.bmi());
        /*int age = 19;
        Integer age2 = 19;
        System.out.println(age2.equals(19));
        char a = 'A';
        float f = 19.7f;
        byte b = 120;
        boolean adult = true;
        String name = "Apple";*/


    }
}

File: src/main/kotlin/com/tom/kotlin/Hello.kt
package com.tom.kotlin

fun main() {
//    println("Hello World!")

    var s : String? = "abcde"
    s = null
    println(s?.length)
    println(s?.get(3))
    println(s?.substring(3))

    val h = Human(weight = 45f,height = 1.6f)
    println(h.bmi())
/*    val age : Int = 19; //val 表示固定不可改變
    var age2 = 20; // var可以被改變
    age2 = 21;*/

//    val n = Human("JB",weight = 50f,height = 1.7f)
}

class Human(var weight: Float, var height: Float, var name: String = ""){ //變數可給預設值

    init {
        println("test weight = $weight") //加了"$"可直接使用變數
    }

    constructor(name: String, weight: Float, height: Float) : this(weight, height, name) {
        //這是次要建構子
        //建構子中不可用var、val定義變數
        println("secondary constructor = $name" )
    }

    fun bmi() : Float{
        val bmi = weight / (height*height)
        return bmi
    }

    fun hello() { //不回傳資料時，不用寫void
        println("Hello World!")
    }
}